dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -386,6 +379,43 @@ public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, b
         }
     }
 
+    /**
+     * Appends the #workflow/ prefix to the start of the provided path if necessary
+     *
+     * @param entryPath path to either a tool or workflow
+     */
+    @Override
+    public String getTrsId(String entryPath) {
+        return entryPath.startsWith("#workflow/") ? entryPath : "#workflow/" + entryPath;
+    }
+
+    /**
+     * Returns the version ID of the given workflow, falls back to the latest version
+     * @param entryPath Workflow path
+     */
+    @Override
+    public String getVersionID(String entryPath) {
+        final String[] parts = entryPath.split(":");
+
+        final String versionID = parts.length > 1 ? parts[1] : "master";
+
+        final Workflow workflow = getDockstoreWorkflowByPath(parts[0]);
+
+        // ensure workflow has version
+        Optional<WorkflowVersion> first = workflow.getWorkflowVersions().stream().filter(foo -> foo.getName().equalsIgnoreCase(versionID))
+            .findFirst();
+
+        // if no master is present (for example, for hosted workflows), fail over to the latest descriptor
+        if (first.isEmpty()) {
+            first = workflow.getWorkflowVersions().stream().max(Comparator.comparing(WorkflowVersion::getLastModified));
+            first.ifPresent(workflowVersion -> out(
+                "Could not locate workflow with version '" + versionID + "'. Using last modified version '" + workflowVersion.getName()
+                    + "' instead."));
+        }
+
+        return first.isEmpty() ? versionID : first.get().getName();
```

Comment:
I don't grok the entire flow, but you are potentially returning the version id of a non-existent version. Is that going to work further down the line?

Reply:
Looking at the code, I think it's OK; the GA4GH call will fail and you'll exit.